SyntheticsRunTestsTask/task.ts

```diff
@@ -1,47 +1,49 @@
 import * as path from 'path'
 
 import * as task from 'azure-pipelines-task-lib/task'
-import * as util from 'azure-pipelines-tasks-packaging-common/util'
 
-import {synthetics} from '@datadog/datadog-ci'
-
-import {renderResults} from './process-results'
-import {reportCiError} from './report-ci-error'
 import {getReporter, resolveConfig} from './resolve-config'
+import {synthetics} from '@datadog/datadog-ci'
 
 async function run(): Promise<void> {
   task.setResourcePath(path.join(__dirname, 'task.json'))
   synthetics.utils.setCiTriggerApp('azure_devops_task')
 
   const reporter = getReporter()
   const config = await resolveConfig(reporter)
+  const startTime = Date.now()
+
+  let results: synthetics.Result[]
+  let summary: synthetics.Summary
 
   try {
-    const startTime = Date.now()
-    const {results, summary} = await synthetics.executeTests(reporter, config)
-    const resultSummary = renderResults({config, reporter, results, startTime, summary})
-    if (
-      resultSummary.criticalErrors > 0 ||
-      resultSummary.failed > 0 ||
-      resultSummary.timedOut > 0 ||
-      resultSummary.testsNotFound.size > 0
-    ) {
-      task.setResult(task.TaskResult.Failed, `Datadog Synthetics tests failed : ${printSummary(resultSummary)}`)
-    } else {
-      task.setResult(task.TaskResult.Succeeded, `Datadog Synthetics tests succeeded : ${printSummary(resultSummary)}`)
-    }
+    ;({results, summary} = await synthetics.executeTests(reporter, config))
   } catch (error) {
-    if (error instanceof synthetics.CiError) {
-      reportCiError(error, reporter)
-    } else {
-      util.logError(`Internal error: ${String(error)}`)
+    synthetics.utils.reportExitLogs(reporter, config, {error})
+
+    const exitReason = synthetics.utils.getExitReason(config, {error})
+    if (exitReason !== 'passed') {
+      task.setResult(task.TaskResult.Failed, 'Running Datadog Synthetics tests failed.')
     }
-    task.setResult(task.TaskResult.Failed, 'Running Datadog Synthetics tests failed.')
+    return
```

Comment:
I'm not a fan of conditional early return - could we keep the lines 31-46 inside of the `try` section ?

Reply:
That would also allow us to avoid the `let results, summary` and stick with `const`.